src/avx2/stage1.rs

```diff
@@ -47,22 +43,21 @@ unsafe fn check_utf8(
         );
     } else {
         // it is not ascii so we have to do heavy work
-        *previous = avxcheck_utf8_bytes(input.lo, &previous, has_error);
-        *previous = avxcheck_utf8_bytes(input.hi, &previous, has_error);
+        *previous = check_utf8_bytes(input.v0, &previous, has_error);
+        *previous = check_utf8_bytes(input.v1, &previous, has_error);
     }
 }
 
 /// a straightforward comparison of a mask against input. 5 uops; would be
 /// cheaper in AVX512.
 #[cfg_attr(not(feature = "no-inline"), inline(always))]
 fn cmp_mask_against_input(input: &SimdInput, m: u8) -> u64 {
+    #[allow(clippy::cast_possible_wrap, clippy::cast_sign_loss)]
```

Comment:
Something I found out recently: moving this infront of the function would prevent tarpaulin from falsly marking this as a non covered line.

Reply:
nifty! so is this good or should I hoist it up to line 54?

Reply:
It's fine there, we got to go over stuff eventually anyway when we take another crack at coverage :) I just wanted to share it